coderd/workspacebuilds.go

```diff
@@ -1002,12 +1016,23 @@ func (api *API) buildTimings(ctx context.Context, build database.WorkspaceBuild)
 	}
 	for _, t := range agentScriptTimings {
 		res.AgentScriptTimings = append(res.AgentScriptTimings, codersdk.AgentScriptTiming{
-			StartedAt:   t.StartedAt,
-			EndedAt:     t.EndedAt,
-			ExitCode:    t.ExitCode,
-			Stage:       string(t.Stage),
-			Status:      string(t.Status),
-			DisplayName: t.DisplayName,
+			StartedAt:          t.StartedAt,
+			EndedAt:            t.EndedAt,
+			ExitCode:           t.ExitCode,
+			Stage:              string(t.Stage),
+			Status:             string(t.Status),
+			DisplayName:        t.DisplayName,
+			WorkspaceAgentID:   t.WorkspaceAgentID.String(),
+			WorkspaceAgentName: t.WorkspaceAgentName,
+		})
+	}
+	for _, agent := range agents {
```

Comment:
Is the case where no agent(s) are found handled specifically by the frontend?

Reply:
the connection timings will be empty so the FE will not display a bar for it.